src/main/java/de/zalando/aruha/nakadi/repository/db/EventTypeCache.java

```diff
@@ -0,0 +1,128 @@
+package de.zalando.aruha.nakadi.repository.db;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import de.zalando.aruha.nakadi.domain.EventType;
+import de.zalando.aruha.nakadi.exceptions.NoSuchEventTypeException;
+import de.zalando.aruha.nakadi.repository.EventTypeRepository;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.cache.PathChildrenCache;
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheEvent;
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheListener;
+import org.apache.curator.utils.ZKPaths;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+
+import java.util.concurrent.ExecutionException;
+
+public class EventTypeCache {
+
+    public final String ZKNODE_PATH = "/nakadi/event_types";
+
+    private final LoadingCache<String, EventType> cache;
+    private final PathChildrenCache cacheSync;
+    private final CuratorFramework zkClient;
+
+    public EventTypeCache(final EventTypeRepository dbRepo, final CuratorFramework zkClient) throws Exception {
+        initParentCacheZNode(zkClient);
+
+        this.zkClient = zkClient;
+        this.cache = setupInMemoryCache(dbRepo);
+        this.cacheSync = setupCacheSync(zkClient);
+    }
+
+    public void updated(final String name) throws Exception {
+        final String path = getZNodePath(name);
+        zkClient.setData().forPath(path, new byte[0]);
+    }
+
+    public EventType get(final String name) throws NoSuchEventTypeException, ExecutionException {
+        try {
+            return cache.get(name);
+        } catch (ExecutionException e) {
+            if (e.getCause() instanceof NoSuchEventTypeException) {
+                NoSuchEventTypeException noSuchEventTypeException = (NoSuchEventTypeException) e.getCause();
+                throw noSuchEventTypeException;
+            } else {
+                throw e;
+            }
+        }
+    }
+
+    public void created(final EventType eventType) throws Exception {
+        final String path = getZNodePath(eventType.getName());
+        zkClient
+                .create()
+                .creatingParentsIfNeeded()
+                .withMode(CreateMode.PERSISTENT)
+                .forPath(path, new byte[0]);
+    }
+
+    public void removed(final String name) throws Exception {
+        final String path = ZKPaths.makePath(ZKNODE_PATH, name);
+        zkClient.delete().forPath(path);
+    }
+
+    private void initParentCacheZNode(final CuratorFramework zkClient) throws Exception {
```

Comment:
I shouldn't be using `creatingParentsIfNeeded()` everywhere, since the node `/nakadi/event_types` should be created only on application bootstrap. This node should be created before the initialisation of the remote cache and never afterwards.